File: exercises/D402/solution.cpp
#include <iomanip>
#include <iostream> 
using namespace std;
long long barcode[100];
double price[100];
int main() {
    int n, m;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> barcode[i] >> price[i];
    }
    double sum = 0.0;
    cin >> m;
    for (int i = 0; i < m; i++) {
        long long k;
        cin >> k;
        for (int j = 0; j < n; j++) {
            if (barcode[j] == k) {
                sum += price[j];
            }
        }
    }
    cout << fixed << setprecision(1) << sum << endl;
}

File: exercises/D402/solution.java
import java.util.*;

class D402 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        long[] barcode = new long[n];
        double[] price = new double[n];
        for (int i = 0; i < n; i++) {
            barcode[i] = in.nextLong();
            price[i] = in.nextDouble();
        }
        double sum = 0;
        int m = in.nextInt();
        for (int i = 0; i < m; i++) {
            long k = in.nextLong();
            for (int j = 0; j < n; j++) {
                if (barcode[j] == k) {
                    sum += price[j];
                }
            }
        }
        System.out.printf("%.1f\n", sum);
    }
}
